### PR TITLE
Geo-location 

### PR DESCRIPTION
**Pull request by faraznaeem**
### PT Link: https://www.pivotaltracker.com/story/show/150355058 ###

- Lookup and added Geo-location to restaurant on create

- Added Gmap to the landing page with the current user location. 

## Thing we learned: ##

- Testing with javascript and cucumber

- Working with geo-location with gmaps.js

- Working with the gem geocoder.

screenshot_2017-08-23-14-08-27 814

### CODE REVIEW

**Comment by tochman** [comment]
You need to pull from upstream/develop

**Comment by faraznaeem** [comment]
Merge with develop branch done :)

**Comment by tochman** [comment]
Right, the last step will be to fix the merge conflicts on this one.

File: features/filtered_location.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    | name             | address1           | postal_code | phone_number | city      |
+    | Vapiano          | Sturegatan 12      | 114 36      | 08 662 20 11 | Stockholm |
+    | Rice             | Nybrogatan 16      | 114 39      | 08-661 32 35 | Stockholm |
+    | Green & Rum      | Östra Hamngatan 16 | 411 09      | 031-15 09 00 | Göteborg  |
+    | Tandoori Kitchen | Friggagatan 15     | 411 01      | 031-23 09 19 | Göteborg  |
+
+@javascript
+Scenario:
+  When I visit "landing" page
+  And my current location is set to "59.33066" lat and "18.057932" lng
```
Comment (tochman) [inline features/filtered_location.feature:16]: We probably need an assertion on this scenario.

Comment (tochman) [reply features/filtered_location.feature:16]: But this is a WIP scenario, right. Will be continued on in next PR?

File: features/location.feature [hunk 2]
```diff
@@ -11,7 +11,7 @@ Background:
...
   And I visit "landing" page
 
-  @javascript
-  Scenario:
-    Then the map has been loaded
-    And my location is set to "59.33066" lat and "18.057932" lng
+@javascript
+Scenario:
+  Then the map has been loaded
+  And my location is set to "59.33066" lat and "18.057932" lng
```
Comment (tochman) [inline features/location.feature]: Well, we will need an assertion here....

File: features/filtered_location.feature (hunk 1 above)
Comment (faraznaeem) [reply features/filtered_location.feature:16]: Yes

**Review by tochman — APPROVED** [review]
A bit messy but we'll go with his now. The important thing is to draw some conclusions from this process and NOT repeat the same mistakes. Good job.

**Pull request merged**